libs/langgraph/src/func.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseCheckpointSaver } from "@langchain/langgraph-checkpoint";
+import { Pregel } from "./pregel/index.js";
+import { PregelNode } from "./pregel/read.js";
+import { END, START } from "./graph/graph.js";
+import { BaseChannel } from "./channels/base.js";
+import { ChannelWrite, PASSTHROUGH } from "./pregel/write.js";
+import { TAG_HIDDEN } from "./constants.js";
+import { ManagedValueSpec } from "./managed/base.js";
+import { EphemeralValue } from "./channels/ephemeral_value.js";
+import { call, getRunnableForFunc } from "./pregel/call.js";
+import { RetryPolicy } from "./pregel/utils/index.js";
+import { Promisified } from "./pregel/types.js";
+
+/**
+ * Options for the @see task function
+ */
+export type TaskOptions = {
+  /**
+   * The retry policy for the task
+   */
+  retry?: RetryPolicy;
+};
+
+/**
+ * Wraps a function in a task that can be retried
+ *
+ *  !!! warning "Experimental"
+ *      This is an experimental API that is subject to change.
+ *      Do not use for production code.
+ *
+ * @param name - The name of the task, analagous to the node name in @see StateGraph
+ * @param func - The function that executes this task
+ * @param options.retry - The retry policy for the task
+ * @returns A proxy function that accepts the same arguments as the original and always returns the result as a @see Promise.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function task<FuncT extends (...args: any[]) => any>(
+  name: string,
+  func: FuncT,
+  options?: TaskOptions
+): (...args: Parameters<FuncT>) => Promisified<FuncT> {
+  return (...args: Parameters<FuncT>) => {
+    const ret = call({ func, name, retry: options?.retry }, ...args);
+    // eslint-disable-next-line no-instanceof/no-instanceof
+    if (ret instanceof Promise) {
```

Comment:
I think this check should be in call, no?

Reply:
This is a bit weird? Why can't we just wrap if it doesn't return a promise?